configs/defs/rocky.textproto

```diff
@@ -1384,3 +1384,4017 @@ benchmark_configs: {
       "}}"
   }
 }
+benchmark_configs: {
+  id: "squashfs-mounting-disabled-rocky"
+  compliance_note: {
+    version: { cpe_uri: "cpe:/o:rockylinux:rockylinux:8.5" version: "1.0.0" benchmark_document: "CIS Rocky Linux 8" }
+    title: "Ensure mounting of squashfs filesystems is disabled"
+    description:
+      "The squashfs filesystem type is a compressed read-only Linux "
+      "filesystem embedded in small footprint systems (similar to cramfs ). A "
+      "squashfs image can be used without having to first decompress the image."
+    rationale:
+      "Removing support for unneeded filesystem types reduces the local attack "
+      "surface of the server. If this filesystem type is not needed, disable "
+      "it."
+    remediation:
+      "Edit or create a file in the /etc/modprobe.d/ directory ending in .conf "
+      "with the lines that read install squashfs /bin/false and blacklist squashfs.\n"
+      "Example: \n"
+      "```\n"
+      "printf \"install squashfs /bin/false "
+      "blacklist squashfs "
+      "\" >> /etc/modprobe.d/squashfs.conf "
+      "```\n"
+      "Run the following command to unload the squashfs module:\n"
+      "```\n"
+      "# rmmod squashfs\n"
+      "```\n"
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "scan_type_specific:{instance_scanning:{"
+      "  check_alternatives:{"
+      "    file_checks:{"
+      "      files_to_check:{single_file:{path:\"/proc/modules\"}}"
+      "      content_entry:{"
+      "        match_type: NONE_MATCH"
+      "        match_criteria: {"
+      "          filter_regex: \"squashfs\\\\s+.*\""
+      "          expected_regex: \"squashfs .*\""
+      "        }"
+      "      }"
+      "    }"
+      "    file_checks:{"
+      "      files_to_check:{files_in_dir:{"
+      "        dir_path:\"/etc/modprobe.d\""
+      "        filename_regex: \".*\\\\.conf\""
+      "        recursive: true"
+      "      }}"
+      "      content_entry:{"
+      "        match_type: ALL_MATCH_ANY_ORDER"
+      "        match_criteria: {"
+      "          filter_regex: \"install squashfs .*\""
+      "          expected_regex: \"install squashfs /bin/(true|false)\""
+      "        }"
+      "      }"
+      "    }"
+      "  }"
+      "  check_alternatives:{"
+      "    file_checks:{"
+      "      files_to_check:{single_file:{path:\"/etc/modprobe.d\"}}"
```

Comment:
I believe we kept these additional checks from the "template" controls we referred to. Is there a specific reason (i.e., performance or code conciseness) for which it would be preferable to remove them?

Reply:
It's mostly for code conciseness, I don't think there's a big performance impact of having the additional checks. I'm not sure what you mean by template controls though?

Reply:
When writing the new benchmarks, we referred to existing ones with similar content or structure from other config files if available. In this case the one we used probably had that check already built in and there was no evident reason to remove it.